aries-site/src/examples/templates/dashboards/content/chartCards/RulesAudit.js

```diff
@@ -0,0 +1,191 @@
+      { result: 'error', count: 0, percent: 0, color: 'status-warning' },
+    ];
+    const ruleMap = new Map();
+    nextAuditResults.forEach(result => {
+      const index = nextAuditResults.findIndex(
+        el => el.result === result.result,
+      );
+      ruleMap.set(result.result, index);
+    });
+    rules.forEach(rule => {
+      const index = ruleMap.get(rule.result);
+      nextAuditResults[index] = {
+        ...nextAuditResults[index],
+        count: nextAuditResults[index].count + 1,
+        percent: ((nextAuditResults[index].count + 1) / rules.length) * 100,
+      };
+    });
+    setAuditResults(nextAuditResults);
+  }, [rules]);
+
+  const grid = {
+    columns: ['auto', 'auto', 'auto'],
+    rows: ['auto', 'auto'],
+    areas: [
+      ['rules', 'frameworks', 'audits'],
+      ['chart', 'chart', 'chart'],
+    ],
+  };
+
+  const nextScheduledAudit = nextAudit && (
+    <Box direction="row" gap="xsmall">
+      <Box
+        // pad aligns icon and text first row
+        pad={{ top: 'xsmall' }}
+      >
+        <Calendar size="small" color="text-strong" />
+      </Box>
+      <Text size="small" weight="bold">
+        Next audit will be performed in{' '}
+        {Math.round(
+          (new Date(nextAudit.scheduledDate) - new Date()) /
+            (1000 * 60 * 60 * 24), // milliseconds to days
+        )}{' '}
+        days.
+      </Text>
+    </Box>
+  );
+
+  return (
+    <ChartCard title="Compliance" subtitle={period} footer={nextScheduledAudit}>
+      <Grid
+        columns={grid.columns}
+        rows={grid.rows}
+        areas={grid.areas}
+        gap="medium"
+      >
+        <Measure
+          gridArea="rules"
+          name={{ label: { label: 'Rules', size: 'medium' } }}
+          value={rules?.length}
+        />
+        <Measure
+          gridArea="frameworks"
+          name={{ label: { label: 'Frameworks', size: 'medium' } }}
+          value={frameworks?.length}
+        />
+        <Measure
+          gridArea="audits"
+          name={{ label: { label: 'Audits', size: 'medium' } }}
+          value={completedAudits?.length}
+        />
+        <AuditResultsChart gridArea="chart" data={auditResults} />
+      </Grid>
+    </ChartCard>
+  );
+};
+
+RulesAudit.propTypes = {
+  period: PropTypes.oneOf(['Last 30 Days', 'Last Year', 'Lifetime']),
+};
+
+const AuditResultsChart = ({ data, ...rest }) => (
+  <MeterGroup {...rest}>
+    {data.map(datum => (
+      <Fragment key={datum.result}>
+        <Text>{datum.result}</Text>
+        <Meter
+          alignSelf="center"
+          type="bar"
+          values={[
+            {
+              value: datum.percent,
+              color: datum.color,
+              label: datum.result,
+            },
+          ]}
+          thickness="small"
+        />
+        <Box align="end">
+          <Text weight="bold">{datum.count}</Text>
+        </Box>
+      </Fragment>
+    ))}
+    <Box />
```

Comment:
can we leave comments around why we need empty boxes?

Reply:
Sure.

Reply:
Done.